src/test/ui/borrowck/mutability-errors.stderr

```diff
@@ -119,146 +119,96 @@ LL |     &mut (*f()).0;
+   |                       - change this to accept `FnMut` instead of `Fn`
 ...
-LL |       fn_ref(move || {
-   |  _____------_-
-   | |     |
-   | |     expects `Fn` instead of `FnMut`
-LL | |         x = (1,);
-   | |         ^^^^^^^^ cannot assign
-LL | |         x.0 = 1;
-LL | |         &mut x;
-LL | |         &mut x.0;
-LL | |     });
-   | |_____- in this closure
+LL |     fn_ref(move || {
+   |     ------ expects `Fn` instead of `FnMut`
+LL |         x = (1,);
+   |         ^^^^^^^^ cannot assign
 
 error[E0594]: cannot assign to `x.0`, as `Fn` closures cannot mutate their captured variables
   --> $DIR/mutability-errors.rs:47:9
    |
-LL |   fn fn_ref<F: Fn()>(f: F) -> F { f }
-   |                         - change this to accept `FnMut` instead of `Fn`
+LL | fn fn_ref<F: Fn()>(f: F) -> F { f }
+   |                       - change this to accept `FnMut` instead of `Fn`
 ...
-LL |       fn_ref(move || {
-   |  _____------_-
-   | |     |
-   | |     expects `Fn` instead of `FnMut`
-LL | |         x = (1,);
-LL | |         x.0 = 1;
-   | |         ^^^^^^^ cannot assign
-LL | |         &mut x;
-LL | |         &mut x.0;
-LL | |     });
-   | |_____- in this closure
+LL |     fn_ref(move || {
+   |     ------ expects `Fn` instead of `FnMut`
+LL |         x = (1,);
+LL |         x.0 = 1;
+   |         ^^^^^^^ cannot assign
 
 error[E0596]: cannot borrow `x` as mutable, as it is a captured variable in a `Fn` closure
   --> $DIR/mutability-errors.rs:48:9
    |
-LL |   fn fn_ref<F: Fn()>(f: F) -> F { f }
-   |                         - change this to accept `FnMut` instead of `Fn`
+LL | fn fn_ref<F: Fn()>(f: F) -> F { f }
+   |                       - change this to accept `FnMut` instead of `Fn`
+...
+LL |     fn_ref(move || {
+   |     ------ expects `Fn` instead of `FnMut`
 ...
-LL |       fn_ref(move || {
-   |  _____------_-
-   | |     |
-   | |     expects `Fn` instead of `FnMut`
-LL | |         x = (1,);
-LL | |         x.0 = 1;
-LL | |         &mut x;
-   | |         ^^^^^^ cannot borrow as mutable
-LL | |         &mut x.0;
-LL | |     });
-   | |_____- in this closure
+LL |         &mut x;
+   |         ^^^^^^ cannot borrow as mutable
 
 error[E0596]: cannot borrow `x.0` as mutable, as `Fn` closures cannot mutate their captured variables
   --> $DIR/mutability-errors.rs:49:9
    |
-LL |   fn fn_ref<F: Fn()>(f: F) -> F { f }
-   |                         - change this to accept `FnMut` instead of `Fn`
+LL | fn fn_ref<F: Fn()>(f: F) -> F { f }
+   |                       - change this to accept `FnMut` instead of `Fn`
+...
+LL |     fn_ref(move || {
+   |     ------ expects `Fn` instead of `FnMut`
 ...
-LL |       fn_ref(move || {
-   |  _____------_-
-   | |     |
-   | |     expects `Fn` instead of `FnMut`
-LL | |         x = (1,);
-LL | |         x.0 = 1;
-LL | |         &mut x;
-LL | |         &mut x.0;
-   | |         ^^^^^^^^ cannot borrow as mutable
-LL | |     });
-   | |_____- in this closure
```

Comment:
This looks like a regression to me. We are not pointing at the closure header. We probably have some extra logic to see if the closure is in a single line that is imperfect and now doesn't trigger even when it should.